File: mobile-ui/src/androidTest/java/org/buffer/android/boilerplate/ui/injection/component/TestApplicationComponent.java
package org.buffer.android.boilerplate.ui.injection.component;

import android.app.Application;

import org.buffer.android.boilerplate.domain.executor.PostExecutionThread;
import org.buffer.android.boilerplate.domain.repository.BufferooRepository;
import org.buffer.android.boilerplate.ui.injection.ApplicationComponent;
import org.buffer.android.boilerplate.ui.injection.module.ActivityBindingModule;
import org.buffer.android.boilerplate.ui.injection.scopes.PerApplication;
import org.buffer.android.boilerplate.ui.TestApplication;
import org.buffer.android.boilerplate.ui.injection.module.TestApplicationModule;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

@Component(modules = {
        TestApplicationModule.class, ActivityBindingModule.class, AndroidSupportInjectionModule.class
})
@PerApplication
public interface TestApplicationComponent extends ApplicationComponent {

    BufferooRepository bufferooRepository();

    PostExecutionThread postExecutionThread();

    void inject(TestApplication application);

    @Component.Builder
    interface Builder {
        @BindsInstance
        TestApplicationComponent.Builder application(Application application);

        TestApplicationComponent build();
    }

}

File: mobile-ui/src/androidTest/java/org/buffer/android/boilerplate/ui/injection/component/TestApplicationComponent.kt
package org.buffer.android.boilerplate.ui.injection.component

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import org.buffer.android.boilerplate.domain.executor.PostExecutionThread
import org.buffer.android.boilerplate.domain.repository.BufferooRepository
import org.buffer.android.boilerplate.ui.injection.ApplicationComponent
import org.buffer.android.boilerplate.ui.injection.module.ActivityBindingModule
import org.buffer.android.boilerplate.ui.injection.module.TestApplicationModule
import org.buffer.android.boilerplate.ui.injection.scopes.PerApplication
import org.buffer.android.boilerplate.ui.test.TestApplication

@Component(modules = arrayOf(TestApplicationModule::class, ActivityBindingModule::class,
        AndroidSupportInjectionModule::class))
@PerApplication
interface TestApplicationComponent : ApplicationComponent {

    fun bufferooRepository(): BufferooRepository

    fun postExecutionThread(): PostExecutionThread

    fun inject(application: TestApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): TestApplicationComponent.Builder

        fun build(): TestApplicationComponent
    }

}